src/ocean/net/http/TaskHttpConnectionHandler.d

```diff
@@ -0,0 +1,440 @@
+/******************************************************************************
+
+    HTTP connection handler base class for use with the SelectListener. The
+    connection and request handler methods are run in a task.
+
+    To build a HTTP server, create a `TaskHttpConnectionHandler` subclass which
+    implements `handleRequest()`, and use that subclass as connection handler in
+    the `SelectListener`.
+
+    Copyright:
+        Copyright (c) 2017 Sociomantic Labs GmbH.
+        All rights reserved.
+
+    License:
+        Boost Software License Version 1.0. See LICENSE_BOOST.txt for details.
+        Alternatively, this file may be distributed under the terms of the Tango
+        3-Clause BSD License (see LICENSE_BSD.txt for details).
+
+ ******************************************************************************/
+
+module ocean.net.http.TaskHttpConnectionHandler;
+
+/******************************************************************************
+
+    Imports
+
+ ******************************************************************************/
+
+import ocean.transition;
+
+import ocean.net.server.connection.TaskConnectionHandler;
+
+/******************************************************************************/
+
+abstract class TaskHttpConnectionHandler : TaskConnectionHandler
+{
+    import ocean.net.http.HttpRequest;
+    import ocean.net.http.HttpResponse;
+    import ocean.net.http.HttpException;
+
+    import ocean.net.http.HttpConst: HttpResponseCode;
+    import ocean.net.http.consts.HttpMethod: HttpMethod;
+    import ocean.net.http.consts.HeaderFieldNames;
+
+    import ocean.io.select.protocol.generic.ErrnoIOException: IOError;
+
+    import ocean.sys.socket.AddressIPSocket;
+    import ocean.sys.ErrnoException;
+    import ocean.core.Enforce;
+
+
+    /**************************************************************************
+
+        HTTP request message parser and response message generator
+
+     **************************************************************************/
+
+    protected HttpRequest   request;
+    protected HttpResponse  response;
+
+    /**************************************************************************
+
+        Reused exception instance; may be thrown by a subclass as well.
+
+     **************************************************************************/
+
+    protected HttpException             http_exception;
```

Comment:
(Not reviewing the rest of the code in this module as I presume it's just copied from the existing HTTP connection handler.)